### PR TITLE
Workaround qcomm resolve attachment bug

### PR DESCRIPTION
**Pull request by cwfitzgerald**
**Connections**

Bug in rend3

**Description**

**Testing**

This is a robust solution for a hack that is proven to work

### CODE REVIEW

File: wgpu-hal/src/vulkan/device.rs [hunk 1]
```diff
@@ -136,6 +136,15 @@ impl super::DeviceShared {
                         .pipeline_bind_point(vk::PipelineBindPoint::GRAPHICS)
                         .color_attachments(&color_refs)
                         .resolve_attachments(&resolve_refs);
+
+                    if self
+                        .workarounds
+                        .contains(super::Workarounds::EMPTY_RESOLVE_ATTACHMENT_LISTS)
+                        && vk_subpass.color_attachment_count == 0
+                    {
+                        vk_subpass.p_resolve_attachments = 0 as _;
```
Comment (kvark) [inline wgpu-hal/src/vulkan/device.rs:145]: interesting choice, I'd expect `ptr::null()` instead

File: wgpu-hal/src/vulkan/device.rs [hunk 2]
```diff
@@ -136,6 +136,15 @@ impl super::DeviceShared {
                         .pipeline_bind_point(vk::PipelineBindPoint::GRAPHICS)
                         .color_attachments(&color_refs)
                         .resolve_attachments(&resolve_refs);
+
+                    if self
+                        .workarounds
+                        .contains(super::Workarounds::EMPTY_RESOLVE_ATTACHMENT_LISTS)
+                        && vk_subpass.color_attachment_count == 0
```
Comment (kvark) [inline wgpu-hal/src/vulkan/device.rs:143]: could simpler check `resolve_refs.is_empty()`

File: wgpu-hal/src/vulkan/mod.rs [hunk 3]
```diff
@@ -172,6 +172,8 @@ bitflags::bitflags!(
     pub struct Workarounds: u32 {
         /// Only generate SPIR-V for one entry point at a time.
         const SEPARATE_ENTRY_POINTS = 0x1;
+        /// Qualcomm OOMs when there are zero color attachments but a non-null pointer to a subpass resolve attachment array. This nulls out that pointer in that case.
```
Comment (kvark) [inline wgpu-hal/src/vulkan/mod.rs:175]: please split the comment into multiple lines for the ease of reading

File: wgpu-hal/src/vulkan/device.rs (hunk 1 above)
Comment (cwfitzgerald) [reply wgpu-hal/src/vulkan/device.rs:145]: Forgot it existed 😅

**Review by kvark — APPROVED** [review]

**Comment by cwfitzgerald** [comment]
Filed https://developer.qualcomm.com/forum/qdn-forums/software/adreno-gpu-sdk/68949 as an upstream issue.

**Pull request merged**